### ISSUE TITLE
Fix S4004 FP: Should not raise on [Serializable] objects

### ISSUE THREAD

**Issue by Corniel** [SonarSource/sonar-dotnet#2762]
### Description
Classes designed for serialization with the `[Serializable]` attribute, require public setters for all there properties, also those who are advised not to be so by S4004. Therefor, S4004 should not apply on `[Serializable]` classes, and this should be explained on the help page.

### Repro steps
``` C#
[Serializable]
public class S4004Serializable
{
    public List<int> Values { get; set; } // Compliant, [Serializable] requires a public setter.
}
```
### Related information
* SonarC# Version 8.0.0

**Comment by pavel-mikula-sonarsource** [comment]
Rule S4004 is not included in profile Sonar way
